lib/generators/declare_schema/migration/migrator.rb

```diff
@@ -478,17 +478,25 @@ def change_indexes(model, old_table_name, to_rename)
           [Array(change_primary_key) + drop_indexes + add_indexes]
         end
 
-        def index_changes_due_to_column_renames(indexes_to_drop, indexes_to_add, to_rename)
-          indexes_to_drop.each_with_object([[], []]) do |index_to_drop, (renamed_indexes_to_drop, renamed_indexes_to_add)|
+        # @return [Array<Array<IndexDefinition>>] pair of arrays of indexes that changed solely due to column renames...not any settings changes.
+        # Note: if the column order changed, that is considered a change of settings--not solely a column rename--so it will not be returned
+        def index_changes_solely_due_to_column_renames(indexes_to_drop, indexes_to_add, to_rename)
+          renamed_indexes_to_drop = []
+          renamed_indexes_to_add = []
+
+          indexes_to_drop.each do |index_to_drop|
             renamed_columns = index_to_drop.columns.map do |column|
               to_rename.fetch(column, column)
-            end.sort
+            end
 
-            if (index_to_add = indexes_to_add.find { |index_to_add| renamed_columns == index_to_add.columns.sort })
+            if (index_to_add = indexes_to_add.find { |index_to_add| renamed_columns == index_to_add.columns }) &&
+                index_to_add.settings == index_to_drop.settings
```

Comment:
Here's the actually bug fix that the `settings` have to be the same. (`settings` is like `options` but doesn't include the name.)